scipy/stats/_distn_infrastructure.py

```diff
@@ -2662,53 +2671,25 @@ def __init__(self, a=0, b=inf, name=None, badvalue=None,
-            decreasing_keys = sorted(self.F.keys(), reverse=True)
-            self.Finv = dict((self.F[k], k) for k in decreasing_keys)
-            self._ppf = instancemethod(vectorize(_drv_ppf, otypes='d'),
-                                       self, rv_discrete)
-            self._pmf = instancemethod(vectorize(_drv_pmf, otypes='d'),
-                                       self, rv_discrete)
-            self._cdf = instancemethod(vectorize(_drv_cdf, otypes='d'),
-                                       self, rv_discrete)
-            self._nonzero = instancemethod(_drv_nonzero, self, rv_discrete)
-            self.generic_moment = instancemethod(_drv_moment,
-                                                 self, rv_discrete)
-            self.moment_gen = instancemethod(_drv_moment_gen,
+            raise ValueError("rv_discrete.__init__(..., values != None, ...)")
+
+        self._construct_argparser(meths_to_inspect=[self._pmf, self._cdf],
+                                  locscale_in='loc=0',
+                                  # scale=1 for discrete RVs
+                                  locscale_out='loc, 1')
+
+        # nin correction needs to be after we know numargs
+        # correct nin for generic moment vectorization
+        _vec_generic_moment = vectorize(_drv2_moment, otypes='d')
+        _vec_generic_moment.nin = self.numargs + 2
+        self.generic_moment = instancemethod(_vec_generic_moment,
                                              self, rv_discrete)
 
-            self.shapes = ' '   # bypass inspection
-            self._construct_argparser(meths_to_inspect=[self._pmf],
-                                      locscale_in='loc=0',
-                                      # scale=1 for discrete RVs
-                                      locscale_out='loc, 1')
-        else:
-            self._construct_argparser(meths_to_inspect=[self._pmf, self._cdf],
-                                      locscale_in='loc=0',
-                                      # scale=1 for discrete RVs
-                                      locscale_out='loc, 1')
-
-            # nin correction needs to be after we know numargs
-            # correct nin for generic moment vectorization
-            _vec_generic_moment = vectorize(_drv2_moment, otypes='d')
-            _vec_generic_moment.nin = self.numargs + 2
-            self.generic_moment = instancemethod(_vec_generic_moment,
-                                                 self, rv_discrete)
-
-            # correct nin for ppf vectorization
-            _vppf = vectorize(_drv2_ppfsingle, otypes='d')
-            _vppf.nin = self.numargs + 2  # +1 is for self
-            self._ppfvec = instancemethod(_vppf,
-                                          self, rv_discrete)
+        # correct nin for ppf vectorization
+        _vppf = vectorize(_drv2_ppfsingle, otypes='d')
+        _vppf.nin = self.numargs + 2  # +1 is for self
+        self._ppfvec = instancemethod(_vppf,
```

Comment:
can these be added now as methods?
I'm not sure how to set `nin` on the method directly.